draft-ietf-quic-recovery.md

```diff
@@ -571,18 +571,22 @@ prior unacknowledged packets to be marked as lost. When an acknowledgement
 is received that newly acknowledges packets, loss detection proceeds as
 dictated by packet and time threshold mechanisms; see {{ack-loss-detection}}.
 
-## Retry and Version Negotiation
-
-A Retry or Version Negotiation packet causes a client to send another Initial
-packet, effectively restarting the connection process and resetting congestion
-control and loss recovery state, including resetting any pending timers.  Either
-packet indicates that the Initial was received but not processed.  Neither
-packet can be treated as an acknowledgment for the Initial.
-
-The client MAY however compute an RTT estimate to the server as the time period
-from when the first Initial was sent to when a Retry or a Version Negotiation
-packet is received.  The client MAY use this value to seed the RTT estimator for
-a subsequent connection attempt to the server.
+## Handling Retry Packets
+
+A Retry packet causes a client to send another Initial packet, effectively
+restarting the connection process.  A Retry packet indicates that the Initial
+was received, but not processed.  A Retry packet cannot be treated as an
+acknowledgment.
+
+Clients that receive a Retry packet reset congestion control and loss recovery
+state, including resetting any pending timers.  Other connection state, in
+particular cryptographic handshake messages, is retained; see Section 17.2.5 of
+{{QUIC-TRANSPORT}}.
+
+The client MAY compute an RTT estimate to the server as the time period from
+when the first Initial was sent to when a Retry or a Version Negotiation packet
+is received.  The client MAY use this value in place of its default for the
+initial RTT estimate.
```

Comment:
If the Retry moves from a server in Melbourne to Chicago, the RTT estimate might not be worth much. Since this could be common use of Retry, I'm not sure this advice is sound. But perhaps I'm missing something?

Reply:
Given how IP routing works, accomplishing what you're suggesting would be quite difficult, but I agree there are cases this isn't perfect.  But it's typically better than any other available value, including the 1 second default.

Reply:
@mikkelfj's point, IIUC, is that connection ID based routing within your server infrastructure could route your new Initial to a server that is far off. I think this is a legitimate use case, but using this sample as the initial RTT value is fine I'd argue, even if it is wrong some times. It is not expected to be the common case, and you replace it in the RTT estimate with the first actual measured value anyways.

Reply:
I tend to agree.  I note that the goal would ordinarily to pick a server that is closer to the client.  In those cases, this will just produce a different overestimation of RTT.  Hopefully it is still less than 1 second.